ocs_ci/ocs/ui/page_objects/buckets_tab.py

```
@@ -632,6 +635,15 @@ def upload_folder_to_bucket(self, folder_path: str, wait_time: int = 2) -> None:
         file_input.clear()
         file_input.send_keys(folder_path)
         time.sleep(wait_time)
+        folder_name = os.path.basename(folder_path)
+        try:
+            self.wait_for_element_to_be_present(
+                (f"//span[contains(text(), '{folder_name}')]", By.XPATH),
+                timeout=timeout,
+            )
+            logger.info(f"Upload verified: {folder_name} appears in bucket")
+        except TimeoutException:
+            raise Exception(f"Upload failed: {folder_name} not found after {timeout}s")
```

Ruff (0.15.20)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)